### ISSUE TITLE
Make this app open-source. 

### ISSUE THREAD

**Issue by ghost** [AChep/keyguard-app#28]
This app looks and works amazing, there are reasons why it would be nice to make this app open-source:

1. Adds more trust in the app, this is a password manager after all and I think a level of trust would be nice <3

2. Allowing users to make pull requests for adding more features and faster bug fixing.

(There are probably other reasons but these are the only two I can think of in the top of my head)

This is just a.. proposal? Recommendation? Whatever, you don't have to make it open-source, but I hope you can take this into consideration! Anyways, great app :)

**Comment by AChep** [comment]
The plan is to release code on Jan 1 under a fairly restrictive license.

**Comment by WreckingBANG** [comment]
@AChep Thank you so much for keeping up on your promise. Thank you so much. I love the Material You Design. Keep Up the good work and i am excited for the first release on github.

**Comment by AChep** [comment]
This is not technically "open source", as the current license basically doesn't allow derivatives. That may change in the future, but for now it will stay as it is.

**Comment by WreckingBANG** [comment]
Yes i know but the Source Code is uploaded. I can now safely use the app. Not that i dont trust you, i just try to avoid closed source stuff and thats why i did not use it until now.

**Comment by elliotrpage** [comment]
Thank you for uploading the source.
Would you be willing to elaborate on your choice of license?
"The source code is available for personal use only." Is not one I have come across often.

**Comment by AChep** [comment]
> Thank you for uploading the source. Would you be willing to elaborate on your choice of license? "The source code is available for personal use only." Is not one I have come across often.

In short, I want Keyguard to be my part time job and not just a hobby. For that it needs to be monetized. I'm afraid that if I use a _proper_ open source license, there will be a fork that just mirrors all the code changed with the paywall cut off. Donations are enough for the supply of cookies, but not close to be even a part time job.

**Comment by danaimone** [comment]
@AChep is there any plans for a iOS version? I was interested in in forking and developing one, but if your intent is to monetize and license is restrictive I don't want to overstep. Love what you're doing with this project, keep it up!

**Comment by AChep** [comment]
> @AChep is there any plans for a iOS version? I was interested in in forking and developing one, but if your intent is to monetize and license is restrictive I don't want to overstep. Love what you're doing with this project, keep it up!

I'm not an iOS developer, but I will give it a try after Compose for iOS goes stable.

**Comment by danaimone** [comment]
> > @AChep is there any plans for a iOS version? I was interested in in forking and developing one, but if your intent is to monetize and license is restrictive I don't want to overstep. Love what you're doing with this project, keep it up!
> 
> I'm not an iOS developer, but I will give it a try after Compose for iOS goes stable.

Sweet!! Neither am I, but I love what you're doing here and it sounds like a fun way to learn 😁

**Comment by flexagoon** [comment]
@AChep have you considered an "eventually open-source" license like BSL or, even better, its improved version, [FSL](https://fsl.software/)?

FSL works like this:

1. It allows people to use the code for any purpose **except creating a competitor to your service**
2. After 2 years, your code becomes fully open source (either Apache or MIT). So people will be able to use your code to create a competing app, but only the code published more than two years ago. Newer code will be available under terms described in point 1.

**Comment by AChep** [comment]
> @AChep have you considered an "eventually open-source" license like BSL or, even better, its improved and simplified version, [FSL](https://fsl.software/)?
> 
> FSL works like this:
> 
> 1. It allows people to use the code for any purpose **except creating a competitor to your service**
> 2. After 2 years, your code becomes fully open source (either Apache or MIT). So people will be able to use your code to create a competing app, but only the code published more than two years ago. Newer code will be available under terms described in point 1.

I did and most likely Keyguard will eventually end up as FSL.

**Comment by flexagoon** [comment]
@AChep that's great to hear, thanks!

**Comment by aloisdg** [comment]
With Bitwarden moving to hostile licence. Is there any update on Keyguard licensing? What about using AGPL to protect the codebase? or the ACSL if you want to block large company?

**Comment by Surferlul** [comment]
Since the Bitwarden Official Clients are only Source Available, this (with Vaultwarden, which is AGPL) would be (imo) the defacto open source way to use Bitwarden. If both Projects are open source, it might also be interesting to advertise Vaultwarden + Keyguard as an entirely open source solution to Password Managing (that additionally is superb). FSL modified to release under AGPL would be absolutely amazing for this purpose.

**Comment by flexagoon** [comment]
> Since the Bitwarden Official Clients are only Source Available

@Surferlul The Bitwarden Android client is GPLv3 licensed

https://github.com/bitwarden/android/blob/main/LICENSE.txt

**Comment by Surferlul** [comment]
@flexagoon The android app are, but the Desktop and web apps aren't. Large parts of the Desktop apps are GPL, but they depend on the internal Bitwarden SDK which isn't.

**Comment by RokeJulianLockhart** [comment]
@Surferlul, I'm the author of [that cited issue](https://github.com/bitwarden/sdk/issues/898#issue-2403231331). I suggest you read https://github.com/bitwarden/sdk/issues/898#issuecomment-2438705858:

> The mobile client is again suitable for inclusion in F-Droid, per [gitlab.com/fdroid/rfp/-/issues/114#note_cf629f7d0a0499cc0e57963e883018da5bfcc712](https://gitlab.com/fdroid/rfp/-/issues/114#note_cf629f7d0a0499cc0e57963e883018da5bfcc712). Shall hide https://github.com/bitwarden/sdk/issues/898#issuecomment-2426022420 as resolved.

Summarily, the proprietary component of the SDK was separated from the rest of the SDK, and now applies solely to the secrets manager. This means that although the SDK remains a dependency of the clients, it is FOSS.

**Comment by Surferlul** [comment]
@RokeJulianLockhart Well I'm glad to have been Proved wrong. The change means that the Desktop clients are also FOSS now, right?

**Comment by RokeJulianLockhart** [comment]
> https://github.com/AChep/keyguard-app/issues/28#issuecomment-2466398582

@Surferlul, to my knowledge. See https://github.com/bitwarden/clients/issues/11611#issuecomment-2436287977 (formatted and spell-checked):

> Thanks again for submitting the concern here. We have made some adjustments to how the SDK code is organized and packaged to allow you to build and run the app with only GPL/OSI licenses included. The `sdk-internal` package references in the clients now come from a new [`sdk-internal repository`](https://github.com/bitwarden/sdk-internal), which follows the licensing model we have historically used for all of our clients (see [`LICENSE_FAQ.md`](https://github.com/bitwarden/server/blob/main/LICENSE_FAQ.md) for more info). The `sdk-internal` reference only uses GPL licenses at this time. If the reference were to include Bitwarden License code in the future, we will provide a way to produce multiple build variants of the client, similar to what we do with web vault client builds.
> 
> The original [SDK repository](https://github.com/bitwarden/sdk) will be renamed to **`sdk-secrets`**, and retains its existing Bitwarden SDK License structure for our Secrets Manager business products. The `sdk-secrets` repository and packages will no longer be referenced from the client apps, since that code is not used there.

**Comment by saihaze** [comment]
I think an AGPL license for the unpaid part and an all-rights-reserved license for paid versions is worth considering

**Comment by SuperCowProducts** [comment]
> This is not technically "open source", as the current license basically doesn't allow derivatives. That may change in the future, but for now it will stay as it is.

Thanks for clarifying, I was a bit confused when I saw 'proprietary' on Flathub.
